### ISSUE TITLE
🐜 Accessibility Bug: Focus issue in the DEI modals

### ISSUE THREAD

**Issue by RM-1978** [GCTC-NTGC/gc-digital-talent#5414]
## ❌ Current behaviour (issue)
When choosing a menu option under the **My employment equity information [Employment equity options section]**, my focus should be placed at the top of the description of each option in the modal.  Instead, I'm placed on the checkbox for the option. This forces me to have to move to the top of the dialog to read the description, and then choose the checkbox if I want to.
### AT configuration
- NVDA, Windows, Edge 
### How to reproduce
1.	Navigate using NVDA, Windows and Edge to the DEI form within the application profile
2.	Navigate using your keyboard to the Employment Equity options 
3.	Choose an EE Group
4.	Read the information in the modal without navigating to the top of the of the modal to read the information.
5.	Check the box. 
### Screen Capture
![#1737 Screen Capture.JPG](https://images.zenhubusercontent.com/625076b8f67c446dd188ffe5/a08ae8d5-75cb-4617-8e90-175bccf2cb8c)
## ✔ Expected behaviour (fix)
Focus is placed at the top of the description of each item instead of the checkbox so that the information on checkbox option can be read/heard. 
### Potential WCAG Considerations (if any): 
[3.2.1 On Focus](https://w3c.github.io/wcag/understanding/on-focus)
### Acceptance Criteria
- [ ] Focus is placed at the top of the modal, and before the copy explaining each EE option instead of the checkbox. 
- [ ]  Users' ability to move focus onto the checkbox is maintained. 
## Additional information (optional)
1. This change will help screen reader users to experience a meaningful sequence that helps to understand what is being asked of them.

**Comment by petertgiles** [comment]
@esizer could you review this one?  I remember some epic discussions about dialogs vs. modals and where the initial focus should be but I don't remember where we landed on those.

**Comment by esizer** [comment]
> @esizer could you review this one? I remember some epic discussions about dialogs vs. modals and where the initial focus should be but I don't remember where we landed on those.

Yeah, so this is sort of subjective but the reasoning behind it is that [users should have focus drawn to this first focusable element in the dialog](https://www.w3.org/WAI/ARIA/apg/example-index/dialog-modal/dialog#:~:text=Initial%20focus%20is%20set%20on%20the%20first%20input%2C%20which%20is%20the%20first%20focusable%20element.).

This is a default only though, if we can determine a better focus target we can change that with the `onOpenAutoFocus` prop on [`Content`](https://www.radix-ui.com/docs/primitives/components/dialog#content). However, this should remain a consistent experience for users so we shouldn't be bouncing focus around on a per dialog basis and it would be best to avoid doing this if possible and document the code on why we are overriding this behaviour.

**Comment by RM-1978** [comment]
Agreed @esizer that the experience should be consistent. Since the intent of the Dialogue box seems to be to share some context information with the user and then ask them to perform a task, I wonder how many story points a fix like this might engender. I will defer to @substrae and @Jerryescandon if consistently bringing user focus to the information that enables them to understand the action(s) they are expected to take in the modal might make it easier to understand expectations and enhance their experience.

**Comment by RM-1978** [comment]
@petertgiles from our conversations in the last stand up or two, I think we have decided to send the focus to the input as @esizer pointed out that it is regular practice. That said, we should think about when sending the focus to the component leaves out context the user needs to understand what is required of them.

**Comment by petertgiles** [comment]
OK, closing this out for now.  Feel free to reopen if there's an something actionable for devs remaining.

**Comment by RM-1978** [comment]
As we said the Devs need to consider when sending to the focus might confuse the screen reader user.
